### PR TITLE
engineering: Apache Hadoop and Big Data

### PR DESCRIPTION
**Pull request by dudaka**
#### What's this PR do?

- Sharing and introducing an article related to Data Engineering
- Checking if people is interested in this topic
- This is the first section in the topic of Hadoop Developer

### CODE REVIEW

File: Engineering/Apache Hadoop and Big Data.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+How can we protect our data from hard disk failure or data loss, data corruption, etc. ? For example, you have a photo of your loved ones and you treasure that picture. 
+In your mind, there is no way you can lose the picture, how would you protect it? You would keep copies of your picture in different places, maybe one in your personal laptop, one copy in Picasa, one copy in your external hard drive, etc. 
+If your laptop crashes you can still get that picture from Picasa or your external hard drive.
+From the idea, we copy each block of data to two more nodes, in other words, we can replicate the block in two more nodes. 
+In total, we have three copies of each block. 
+
+<p align="center">
+  <img width="500" src="images/Apache%20Hadoop%20and%20Big%20Data/replication.png" alt="100 nodes sol">
+</p>
```
Comment (monotykamary) [inline Engineering/Apache Hadoop and Big Data.md:225]: Likewise for here, HTML tags are not possible on Obsidian.

File: Engineering/Apache Hadoop and Big Data.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+Now, you know what Hadoop is and how it can offer an efficient solution to your maximum close price problem against the 1TB data set. So, you can go back to the business and propose Hadoop to solve the problem and to achieve the execution time that your users are expecting. But if you propose a 100 node cluster to your business expect to get some crazy looks, that's the beauty :)). 
+
+You don't need to have a 100 node cluster, we have seen successful Hadoop production environments from small 10 node cluster all the way to 100 to 1000 node cluster. 
+You can simply even start with a 10 node cluster and if you want to reduce the execution time even further, all you have to do is add more nodes to your cluster. That's simple. In other words, Hadoop will horizontally scale.
+
+## History of Hadoop
+
+<p align="center">
+  <img width="500" src="images/Apache%20Hadoop%20and%20Big%20Data/hadoop-history.png" alt="100 nodes sol">
+</p>
```
Comment (monotykamary) [inline Engineering/Apache Hadoop and Big Data.md:257]: Likewise for here, HTML tags are not possible on Obsidian.

**Review by monotykamary — CHANGES_REQUESTED** [review]
Content seems good. The only thing that needs changes are where the image assets are kept and how they are referenced in the markdown for Obsidian.

We keep our image assets in the `_assets` folder. The filenames of every file in Obsidian is **globally availabe**, meaning we reference them by just their name, regardless of where they are kept. In markdown it looks something like this:

With the image location being: ```_assets/100-nodes-sol.png```
```md
...

This is exactly what will happen when 100 nodes trying to transfer the data over the network at the same time.

 ![[100-nodes-sol.png]]

...
```

If you wish to resize the image to a target size, use the following rules and syntax:

> For markdown images, use `AltText|100x100`
For embeds, use `![[image.png|100x100]]`
To have the image scale according to its aspect ratio, omit the height `![[image.png|100]]`

**Review by zlatanpham — APPROVED** [review]
Going to merge the PR. I will format and update the images properly afterward.

**Pull request merged**